src/Data/MessagePack/Class.hs

```diff
@@ -106,8 +106,9 @@ instance MessagePack Object where
   fromObject = return
 
 instance MessagePack () where
-  toObject _ = ObjectArray []
+  toObject _ = ObjectNil
   fromObject = \case
+    ObjectNil      -> return ()
     ObjectArray [] -> return ()
```

Comment:
Add `// TODO(#XXX): Remove in next version" where XXX is the number of the issue you'll create to track the removal of that code.